docs/ref/tasks.txt

```diff
@@ -0,0 +1,318 @@
+=====
+Tasks
+=====
+
+.. module:: django.tasks
+   :synopsis: Django's built-in background task system.
+
+Task definition
+===============
+
+.. function:: task
+
+A decorator defining a :class:`Task`::
+
+    from django.tasks import task
+
+
+    @task
+    def calculate_meaning_of_life() -> int:
+        return 42
+
+In the above example, ``calculate_meaning_of_life`` is a :class:`Task` instance.
+
+``task`` accepts a number of arguments, which are used as-is for the ``Task``:
+
+* :attr:`priority <Task.priority>`
+* :attr:`backend <Task.backend>`
+* :attr:`queue_name <Task.queue_name>`
+* :attr:`enqueue_on_commit <Task.enqueue_on_commit>`
+
+If the Task is not valid, according to the backend,
+:exc:`django.tasks.exceptions.InvalidTaskError` is raised.
+
+.. class:: Task
+
+An class representing a Task to be run in the background. Tasks should be
+defined using the :func:`task` decorator.
+
+Attributes of ``Task`` cannot be modified.
+
+.. attribute:: Task.priority
+
+    The priority of the Task. Priorities must be between -100 and 100, where
+    larger numbers are higher priority, and will be run sooner.
+
+    By default, Tasks are enqueued with a priority of 0.
+
+.. attribute:: Task.backend
+
+    The alias of the backend the Task should be enqueued to.
+
+.. attribute:: Task.queue_name
+
+    The name of the queue the Task will be enqueued on to. Defaults to
+    ``"default"``.
+
+    This must match a queue defined in :setting:`QUEUES <TASKS-QUEUES>`.
+
+.. attribute:: Task.run_after
+
+    The earliest time the Task will be executed, or ``None`` to have no time set.
+
+.. attribute:: Task.enqueue_on_commit
+
+    Whether the Task should be enqueued when the transaction commits successfully,
+    or immediately.
+
+    By default, the behavior of the backend's
+    :setting:`ENQUEUE_ON_COMMIT <TASKS-ENQUEUE_ON_COMMIT>` is used.
+
+.. attribute:: Task.name
+
+    An identifier for a Task. Potentially useful for debugging.
+
+    A Task's name is not necessarily unique.
+
+.. method:: Task.using()
+
+    Create a new Task with modified defaults. The existing Task is left unchanged.
```

Comment:
```suggestion
    Creates a new ``Task`` with modified defaults. The existing ``Task`` is
    left unchanged.
```
(nit) I prefer the tense change and would apply this in a lot of the definitions (Enqueues..., Retrieves...)